api/v1alpha1/phapplication_types.go

```diff
@@ -73,3 +87,64 @@ type PhApplicationList struct {
 func init() {
 	SchemeBuilder.Register(&PhApplication{}, &PhApplicationList{})
 }
+
+func (in *PhApplication) GroupNetworkPolicyIngressRule() []networkv1.NetworkPolicyIngressRule {
+	var ports []networkv1.NetworkPolicyPort
+	groupName := escapism.EscapeToPrimehubLabel(in.Spec.GroupName)
+	for _, p := range in.Spec.SvcTemplate.Spec.Ports {
+		ports = append(ports, networkv1.NetworkPolicyPort{
+			Protocol: &p.Protocol,
+			Port:     &p.TargetPort,
+		})
+	}
+	return []networkv1.NetworkPolicyIngressRule{
+		{
+			Ports: ports,
+			From: []networkv1.NetworkPolicyPeer{
+				{
+					PodSelector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{"primehub.io/group": groupName},
```

Comment:
Can we use `in.GroupName()` instead?

Reply:
Change all the `groupName` in phapplication_types.go to use `in.GroupName()`